test/IntegrationTests/DomainNeutralTests.cs

```diff
@@ -38,6 +38,7 @@ public async Task SubmitsTraces()
         EnvironmentTools.IsWindowsAdministrator().Should().BeTrue();
 
         using var collector = await MockSpansCollector.Start(Output);
+        SetExporter(collector);
```

Comment:
Seeing that this pattern is going to repeat a lot it may be the case that a template pattern applies here. However, because of test method attributes it seems that this wouldn't be that smooth: there will be two method on the test class, the template itself plus a public one just to add the test attributes and call the base class method 🤔 

Not sure if it is worth... any let's leave it outside this PR.

Reply:
I would rather add 3 methods (for each MockCollector) in `TestHelper` that starts a collector and sets it as an exporter.
I can do it in a separate PR.

Then this:

```csharp
        using var collector = await MockSpansCollector.Start(Output);
        SetExporter(collector);
```

would become e.g.

```
        using var collector = await StartSpansCollector();
```

Reply:
100% a separate PR.